index.js

```diff
@@ -0,0 +1,111 @@
+var net = require('net');
+var tls = require('tls');
+var http = require('http');
+var https = require('https');
+var ws = require('ws');
+var _ = require('underscore');
+var url = require('url');
+var wsStream = require('websocket-stream');
+var async = require('async');
+var enableDestroy = require('server-destroy');
+
+function createServer(clientHandler) {
+  return net.createServer(function(client){
+    clientHandler(client);
+  });
+}
+
+function createSecureServer(sslOptions, clientHandler){
+  return tls.createServer(sslOptions, function(client){
+    clientHandler(client);
+  });
+}
+
+function attachServer(server, clientHandler) {
+  (new ws.Server({
+    server: server
+  })).on('connection', function(ws) {
+    clientHandler(wsStream(ws));
+  });
+}
+
+function createWebSocketServer(clientHandler){
+  var server = http.createServer();
+  attachServer(server, clientHandler);
+  return server;
+}
+
+function createSecureWebSocketServer(sslOptions, clientHandler){
+  var server = https.createServer(sslOptions);
+  attachServer(server, clientHandler);
+  return server;
+}
+
+module.exports = function(serverConfig, sharedConfig, clientStreamHandler){
+  if(typeof sharedConfig == 'function') {
+    clientStreamHandler = sharedConfig;
+    sharedConfig = {};
+  }
+
+  if(typeof sharedConfig != 'object') {
+    sharedConfig = {};
+  }
+
+  var servers = {};
+
+  _.each(serverConfig, function(config, id){
+    if(typeof config == 'string') {
+      var c = url.parse(config);
+      config = {
+        protocol: c.protocol.replace(/:$/, ''),
+        port: c.port,
+        host: c.hostname
+      };
+    }
+
+    config = _.defaults(config, sharedConfig, {
```

Comment:
Use [xtend](http://npm.im/xtend) to achieve the same result, or just a bunch of ifs :).